Source/Runtime/Events.Processing/Projections.proto

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+syntax = "proto3";
+
+import "Fundamentals/Artifacts/Artifact.proto";
+import "Fundamentals/Protobuf/Uuid.proto";
+import "Fundamentals/Protobuf/Failure.proto";
+import "Fundamentals/Services/ReverseCallContext.proto";
+import "Fundamentals/Services/Ping.proto";
+import "Runtime/Events.Processing/StreamEvent.proto";
+import "Runtime/Events.Processing/Processors.proto";
+
+package dolittle.runtime.events.processing;
+
+option csharp_namespace = "Dolittle.Runtime.Events.Processing.Contracts";
+
+enum ProjectEventKeySelectorType {
+    EVENT_SOURCE_ID = 0;
+    PARTITION_ID = 1;
+    PROPERTY = 2;
+}
+
+message ProjectionEventSpecification {
+    artifacts.Artifact eventType = 1;
+    ProjectEventKeySelectorType keySelectorType = 2;
+    string keySelectorExpression = 3;
```

Comment:
Turns out we don't support it now (at least for DotNet) - I suggest we leave it as is and consider fixing it later. It doesn't add too much overhead to send an empty string, and it would be a non-breaking change to add that into the Contracts later if we want to.